tests/test_aaa_authentication_login.rb

```diff
@@ -0,0 +1,218 @@
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/aaa_authentication_login'
+
+# Test class for AAA Authentication Login
+class TestAaaAuthenticationLogin < CiscoTestCase
+  DEFAULT_AAA_AUTHENTICATION_LOGIN_ASCII_AUTH = false
+  DEFAULT_AAA_AUTHENTICATION_LOGIN_CHAP_ENABLE = false
+  DEFAULT_AAA_AUTHENTICATION_LOGIN_ERROR_ENABLE = false
+  DEFAULT_AAA_AUTHENTICATION_LOGIN_MSCHAP_ENABLE = false
+  DEFAULT_AAA_AUTHENTICATION_LOGIN_MSCHAPV2_ENABLE = false
+
+  def aaaauthenticationlogin_detach(authlogin)
+    # Reset the device to a clean test state. Note that AAA will raise an error
+    # when disabling an authentication method while a different type is present.
+    s = @device.cmd("show run | i 'aaa authentication login'")
+    if s[/aaa authentication login (\S+) enable/]
+      config("no aaa authentication login #{Regexp.last_match(1)} enable")
+    end
+    authlogin.ascii_authentication = DEFAULT_AAA_AUTHENTICATION_LOGIN_ASCII_AUTH
+    authlogin.error_display = DEFAULT_AAA_AUTHENTICATION_LOGIN_ERROR_ENABLE
+  end
+
+  def test_get_ascii_authentication
+    aaaauthlogin = AaaAuthenticationLogin
+
+    config('no aaa authentication login ascii-authentication')
+    refute(aaaauthlogin.ascii_authentication)
+
+    config('aaa authentication login ascii-authentication')
+    assert(aaaauthlogin.ascii_authentication,
+           'Error: AAA authentication login ascii get with preconfig')
+    aaaauthenticationlogin_detach(aaaauthlogin)
+  end
+
+  def test_get_default_ascii_authentication
+    aaaauthlogin = AaaAuthenticationLogin
+    config('no aaa authentication login ascii-authentication')
+    assert_equal(DEFAULT_AAA_AUTHENTICATION_LOGIN_ASCII_AUTH,
+                 aaaauthlogin.default_ascii_authentication,
+                 'Error: AAA authentication login, default ascii incorrect')
+    aaaauthenticationlogin_detach(aaaauthlogin)
+  end
+
+  def test_set_ascii_authentication
+    aaaauthlogin = AaaAuthenticationLogin
+
+    aaaauthlogin.ascii_authentication = true
```

Comment:
Now that you mention it, I think I actually want to change the DEFAULT below to be false instead. Since I'm specifically checking for the presence and absence of the configuration, I don't want to be assuming the default in either case.